hibernate-core/src/main/java/org/hibernate/persister/entity/NamedQueryLoader.java

```diff
@@ -47,15 +47,15 @@ public NamedQueryLoader(String queryName, EntityPersister persister) {
 	}
 
 	@Override
-	public Object load(Serializable id, Object optionalObject, SharedSessionContractImplementor session, LockOptions lockOptions) {
+	public Object load(Serializable id, Object optionalObject, SharedSessionContractImplementor session, LockOptions lockOptions, Boolean readOnly) {
```

Comment:
Probably needs an overloaded variant with the previous signature:

```java
public Object load(Serializable id, Object optionalObject, SharedSessionContractImplementor session, LockOptions lockOptions) {
```

Reply:
If you inspect the parent interface, i.e. UniqueEntityLoader, I've already added the following default method as following:
`
	default Object load(
			Serializable id,
			Object optionalObject,
			SharedSessionContractImplementor session,
			LockOptions lockOptions) {
		return load(id, optionalObject, session, lockOptions, null);
	}
`
So it seems I don't need to overload the variant any more.

Reply:
The previous signature has been implemented as default interface method in its parent interface to route to the new method which accepts a new parameter and hence generalize the old one and is supposed to be the one to be overridden.

Reply:
you are right. Updated.